event/metrics.go

```diff
@@ -0,0 +1,91 @@
+package event
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// MustRegisterMetrics will register all event related metrics on the given registry.
+// If metrics with the same name already exist no the register this function will panic.
+func MustRegisterMetrics(registry *prometheus.Registry) {
+	registry.MustRegister(publishDuration, publishCounter, processCounter, processDuration)
+}
+
+// SampledMessageHandler will instrument the given MessageHandler returning a new one
+// that samples metrics. These will be `event_process_*` metrics using as `name` the
+// given eventName.
+func SampledMessageHandler(handler MessageHandler, eventName string) MessageHandler {
```

Comment:
Not sure this is the best name, open to alternatives. The idea is that now anyone using the RawSubscription can also instrument metrics easily + provide an eventName and the same API is used by the Subscription (which has a event name).

Reply:
for me the name sounds good